backend/db.py

```diff
@@ -127,8 +127,8 @@ def update_job(
         job: Job,
     ) -> Job:
         with closing(sqlite3.connect(self.path)) as connection:
+            connection.execute("PRAGMA foreign_keys = ON")
```

Comment:
why was this moved out? also why no `connection.row_factory = dict_factory` here?

Reply:
Theres no need to wait until we have a cursor before we execute the pragma, I'm going to refactor this at a later date as we should setup pooling when dealing with other database types such as postgres.

Reply:
nit: in that case I think move the rest of the `connection.execute("PRAGMA foreign_keys = ON")` lines before the cursor for consistency